ts/client/src/client.ts

```diff
@@ -829,6 +830,85 @@ export class MangoClient {
     );
   }
 
+  public async emptyAndCloseMangoAccount(
+    group: Group,
+    mangoAccount: MangoAccount,
+  ): Promise<TransactionSignature> {
+    const instructions: TransactionInstruction[] = [];
+    const healthAccountsToExclude: PublicKey[] = [];
+
+    for (const serum3Account of mangoAccount.serum3Active()) {
+      const serum3Market = group.serum3MarketsMapByMarketIndex.get(
+        serum3Account.marketIndex,
+      )!;
+
+      const openOrders = mangoAccount.serum3.find(
+        (account) => account.marketIndex === serum3Market.marketIndex,
+      )!.openOrders;
+
+      const closeOOIx = await this.serum3CloseOpenOrdersIx(
+        group,
+        mangoAccount,
+        serum3Market.serumMarketExternal,
+      );
+      healthAccountsToExclude.push(openOrders);
+      instructions.push(closeOOIx);
+    }
+
+    for (const perp of mangoAccount.perpActive()) {
+      const perpMarketIndex = perp.marketIndex;
```

Comment:
you want to cancel all perp open orders also

Reply:
user need to cancel them before closing https://pbs.twimg.com/media/FJU_ub3XoA0AEdn.png